komodo-relational/src/main/java/org/komodo/relational/datasource/internal/DatasourceValidationParser.java

```diff
@@ -0,0 +1,300 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ *
+ * See the LEGAL.txt file distributed with this work for information regarding copyright ownership and licensing.
+ *
+ * See the AUTHORS.txt file distributed with this work for a full listing of individual contributors.
+ */
+package org.komodo.relational.datasource.internal;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Stack;
+import javax.xml.parsers.SAXParser;
+import javax.xml.parsers.SAXParserFactory;
+import org.komodo.relational.Messages;
+import org.komodo.relational.datasource.Datasource;
+import org.komodo.spi.repository.Repository.UnitOfWork;
+import org.komodo.spi.repository.Repository.UnitOfWork.State;
+import org.komodo.utils.ArgCheck;
+import org.komodo.utils.KLog;
+import org.xml.sax.Attributes;
+import org.xml.sax.SAXException;
+import org.xml.sax.SAXParseException;
+import org.xml.sax.helpers.DefaultHandler;
+
+/**
+ * The <code>DatasourceValidationParser</code> parses a data source definition file.  
+ * This parser validates a file and returns the names of data sources in the file.
+ */
+public class DatasourceValidationParser {
```

Comment:
There is a lot of common code between this class and `DatasourceParser` class. Both of these parsers parse the same input file in the `UploadDatasourceCommand`. I'm wondering if they can be combined into one parser to eliminate the duplicate code and then the input file can be parsed only once. Wdyt?